### PR TITLE
fix loofah gem security vulnerability

### PR DESCRIPTION
**Pull request by pmcostello**
to fix rails security vulnerabilities

### CODE REVIEW

**Comment by tomas-stefano** [comment]
@pmcostello I am getting the error when running the test.sh:

```
$ ./test.sh
...
...
...
Running Database Schema Load
-------------------
rake aborted!
NoMethodError: undefined method `public_key=' for #<Recaptcha::Configuration:0x007fbd59289128>
```

Is the error happening on yours

**Comment by pmcostello** [comment]
yup i got the same error locally, caused by breaking change in recaptcha gem, support for keys defined in the app directly was dropped from v1.0 
need to add require doting to forms controllers to load env variables

File: Gemfile [hunk 1]
```diff
@@ -18,7 +19,7 @@ gem 'flickraw-cached'
 gem 'rubypants', '~> 0.2.0'
 gem 'rake', '~> 10.3.2'
 gem 'fog'
-gem 'recaptcha', require: 'recaptcha/rails', branch: 'rails3'
+gem 'recaptcha', require: 'recaptcha/rails'
```
[the remark this answers is not in the document]
Comment (pmcostello) [reply Gemfile]: I'll create a card to fix the remaining tech debt but should we deploy this update as it fixes the inherent vulnerability in the loofah gem and all the tests pass successfully, albeit with a lot of noise :)

[the remark this answers is not in the document]
Comment (margOnline) [reply Gemfile]: Happy to approve the pr when I see the card(s) for addressing the tech debt.

**Comment by willhall88** [comment]
Agree with the "deprecation warnings re FactoryBot".

The other "noise" in the test script isn't tech debt necessarily it's just the format of the rspec in the test script -
https://github.com/moneyadviceservice/publify/blob/master/test.sh#L33
Are the output formats useful like this for jenkins or Go? if so then we should keep it and perhaps look into a work around for running locally so there is less output

**Comment by willhall88** [comment]
@pmcostello @margOnline I have pushed up a commit fixing the FactoryBot deprecation warnings

**Comment by margOnline** [comment]
@willhall88 fix looks good. I realise that the other 'noise' are not warnings but it is not helpful information so would be good to make it helpful.

**Comment by willhall88** [comment]
yeah, I agree. my question would be whether the formats used are useful elsewhere and thus put it in an `if` or whether we just simplify the output everywhere. @pmcostello @olsharpe ?

**Comment by osharpe** [comment]
I've pushed commit to add argument to RSpec test to send output to tmp file, don't think the noise its particularly useful for Go or Jenkins, we only really care about the exit code. Otherwise looks good to me! 👍

**Comment by osharpe** [comment]
Might be worth sending the RspecJunit output to `tmp` if its useful for debugging during development. Don't think its really that helpful in the build process as you get any failures in the summary and operationally we only care about the exit code. Otherwise looks good to me.

File: Gemfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (pmcostello) [reply Gemfile]: here's the TP card for the Blog tech debt :)

**Comment by pmcostello** [comment]
i've removed the verbose output from the test script as it's not required for the CI build/test processes
think this is ready to merge now, the remaining tech debt TP card is 9113
thanks to all for your help & input on this PR, Go Team! :)

**Pull request merged**